src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/close_application_request_test.cc

```diff
@@ -0,0 +1,345 @@
+  EXPECT_EQ(result_code, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+
+  const bool success =
+      (*response_message)[strings::msg_params][strings::success].asBool();
+  EXPECT_FALSE(success);
+}
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_IGNORED) {
+  using namespace am;
+  auto message = CreateCloseAppMessage();
+
+  CloseApplicationCommandPtr command(
+      CreateCommand<sdl_rpc_plugin::commands::CloseApplicationRequest>(
+          message));
+  MockAppPtr mock_app(CreateMockApp());
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  EXPECT_CALL(*mock_app, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::eType::HMI_NONE));
+
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).Times(0);
+
+  auto response_message = CreateMessage();
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(_, am::commands::Command::CommandSource::SOURCE_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&response_message), Return(true)));
+  command->Run();
+
+  const auto result_code = static_cast<mobile_apis::Result::eType>(
+      (*response_message)[strings::msg_params][strings::result_code].asInt());
+  EXPECT_EQ(result_code, mobile_apis::Result::IGNORED);
+
+  const bool success =
+      (*response_message)[strings::msg_params][strings::success].asBool();
+  EXPECT_FALSE(success);
+}
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_INVALID_DATA) {
+  using namespace am;
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::eType::HMI_FULL));
+
+  ON_CALL(*am::MockMessageHelper::message_helper_mock(),
+          GetBCActivateAppRequestToHMI(_, _, _, _, _, _))
+      .WillByDefault(
+          Return(std::make_shared<SmartObject>(smart_objects::SmartType_Null)));
+
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).WillOnce(Return(false));
+
+  auto response_message = CreateMessage();
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(_, am::commands::Command::CommandSource::SOURCE_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&response_message), Return(true)));
+
+  command_->Run();
+
+  const auto result_code = static_cast<mobile_apis::Result::eType>(
+      (*response_message)[strings::msg_params][strings::result_code].asInt());
+  EXPECT_EQ(result_code, mobile_apis::Result::INVALID_DATA);
+
+  const bool success =
+      (*response_message)[strings::msg_params][strings::success].asBool();
+  EXPECT_FALSE(success);
+}
+
+TEST_F(CloseApplicationRequestTest,
+       CloseApplicationRequestSendsResponse_GENERIC_ERROR) {
+  using namespace am;
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::eType::HMI_FULL));
+
```

Comment:
@IGapchuk Seems that we can add additional method in class and move this code 
` EXPECT_CALL(app_mngr_, application(kConnectionKey))
      .WillOnce(Return(mock_app_));
  EXPECT_CALL(*mock_app_, hmi_level())
      .WillOnce(Return(mobile_apis::HMILevel::eType::HMI_FULL));
  ON_CALL(*am::MockMessageHelper::message_helper_mock(),
          GetBCActivateAppRequestToHMI(_, _, _, _, _, _))
      .WillByDefault(
          Return(std::make_shared<SmartObject>(smart_objects::SmartType_Null)));
  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).WillOnce(Return(true));`
there
Is it possible?

Reply:
@ZhdanovP Fixed in a465d3a1e8eb9abeaeb8836a4917d23c59be3648.